scene/main/node.cpp

```diff
@@ -1330,7 +1330,7 @@ void Node::_generate_serial_child_name(const Node *p_child, StringName &name) co
 
 	// Assign the base name + separator to name if we have numbers preceded by a separator
 	if (nums.length() > 0 && name_string.substr(name_last_index, nnsep.length()) == nnsep) {
-		name_string = name_string.substr(0, name_last_index + nnsep.length());
+		name_string = name_string.left(name_last_index + nnsep.length());
```

Comment:
This is complex, but in a nutshell, I haven't determined if my change affects any behaviors. But if it does, then I'm 99% sure the old behavior would've also been problematic.